rabbitmq/handlers.go

```diff
@@ -0,0 +1,60 @@
+package rabbitmq
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	amqp "github.com/rabbitmq/amqp091-go"
+)
+
+// PasswordChangeHandler handles password change messages.
+type PasswordChangeHandler struct{}
+
+// NewPasswordChangeHandler creates a new password change handler.
+func NewPasswordChangeHandler() *PasswordChangeHandler {
+	return &PasswordChangeHandler{}
+}
+
+// PasswordChangeMessage represents a password change message.
+type PasswordChangeMessage struct {
+	Domain      string `json:"domain"`
+	NewPassword string `json:"new_password"`
+	Version     int    `json:"version"`
+}
```

Comment:
This does not completely look like the message defined in the ADR (no version in the ADR and missing fields here).